CMakeVOL.cmake

```diff
@@ -34,11 +34,28 @@ if (HDF5_VOL_ALLOW_EXTERNAL)
     message (FATAL_ERROR "HDF5_ALLOW_EXTERNAL_SUPPORT must be set to 'GIT' to allow building of external HDF5 VOL connectors")
   endif ()
 
+  if (BUILD_SHARED_LIBS)
+    set(HDF5_LIB_TARGETS ${HDF5_LIBSH_TARGET})
+    set(HDF5_HL_LIB_TARGETS ${HDF5_HL_LIBSH_TARGET})
+  else ()
+    if (BUILD_STATIC_LIBS)
```

Comment:
As is, we shouldn't ever be trying to use static libraries when building VOL connectors, so I don't think we want these changes